### ISSUE TITLE
FN2 Regression Testing: Apple account onboarding, iCloud, Apple 2FA, etc. stats

### ISSUE THREAD

**Issue by ChristopherA** [BlockchainCommons/GordianWallet-iOS#59]
We need to do some regression testing on various states that a users can be in the first time they are using FN2 and possibly have a new or incompletely established iCloud account, or don't have 2FA set up yet.

This is related to a problem that @watersnake1 had turned of 2FA and didn't know it. He has fixed this personally, but FN2 should not have crash.

Some of these probably may also be difficult to reproduce as the FN2 will be put in the background when Apple does some onboarding.

* [ ] Create list of possible states to test
* [ ] How do we test these states
* [ ] Setup a new Apple user account on phone and test it.
…

**Comment by Fonta1n3** [comment]
Of course more testing will be helpful but I would not assume his crash was caused by anything iCloud or 2fa related. It was most likely a Tor issue, especially if it occurs when app is entering foreground or background (or soon after) which was the case for him I beleive. [This](https://github.com/iCepa/Tor.framework/issues/65) is effecting us.

**Comment by watersnake1** [comment]
For clarity, here are the steps to produce the error:
Sign into iCloud with an account without 2fa enabled. 
Download latest testflight release of FN2.
Start FN2, scroll through intro screens until presented with camera for scanning QR code.
FN2 prompts to enable Apple 2fa, then redirects user to Settings->Apple Id (At the top of settings).
FN2 Crashes.
Cycle repeats if the user does not enable apple 2fa.

One of the problems here is that FN2 does not redirect the user all the way to the 2fa menu in settings (and probably can't, because viewing it requires entering your password). Because I was going off the assumption that I already had 2fa enabled, and there was no indicator otherwise, it took me awhile to realize why this problem was happening. Although it seems obvious now, at the time I did not make the mental connection about what was causing the issue (and lots of users probably wouldn't either). A more specific message / warning before opening the settings app would be helpful I think. 

Also, apple no longer allows for users to disable 2fa on their accounts if it is already enabled. So reproducing the error will require creating a new apple id.

**Comment by Fonta1n3** [comment]
@watersnake1 it sounds like this crash was consistently reproduced, is that correct? Thanks for the info

**Comment by watersnake1** [comment]
Yes it is repeatable as long as you have an Apple ID without 2fa on. Once it gets activated it can’t be turned off.
